attribute-projection-registry/src/main/java/org/hypertrace/core/attribute/service/projection/TriFunction.java

```diff
@@ -0,0 +1,13 @@
+package org.hypertrace.core.attribute.service.projection;
+
+import java.util.Objects;
+import java.util.function.Function;
+
+public interface TriFunction<T, U, V, R> {
+  R apply(T t, U u, V v);
+
+  default <W> TriFunction<T, U, V, W> andThen(Function<? super R, ? extends W> after) {
```

Comment:
just for fun? or are we using andThen anywhere?

Reply:
Oh I copied and modified Java's BiFunction code haha

Reply:
https://media.giphy.com/media/3OfNDMhTCi7zW/giphy.gif